app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,44 @@
+class OrdersController < ApplicationController
+  before_action :authenticate_user!
+
+  load_resource only: :show
+
+  # GET /orders
+  # GET /orders.json
+  def index
+    @orders = current_user.orders
+    @orders = filter_by_status
+  end
+
+  # GET /orders/1
+  # GET /orders/1.json
+  def show
+    if current_user.orders.exists?(id: params[:id])
+      @shipping_a = @order.shipping_address
+      @billing_a = @order.billing_address
+      @delivery = @order.delivery
+      @payment = @order.payment
+      @order_items = @order.order_items
```

Comment:
why do you need setup this variables? why not just using needed association from `@order`?